desc/magnetic_fields.py

```diff
@@ -1819,9 +1836,875 @@ def from_surface(
+                    plt.xlabel("zeta")
+                    plt.title(
+                        f"alpha= {alphas[ialpha_to_plot[whichPlot] - 1]:1.5e}"
+                        + f" index = {ialpha_to_plot[whichPlot] - 1}",
+                        fontsize="x-small",
+                    )
+                    plt.colorbar()
+                    plt.xlim([0, 2 * np.pi / eq.NFP])
+                plt.tight_layout()
+                plt.figtext(
+                    0.5,
+                    0.995,
+                    "Total Current Potential",
+                    horizontalalignment="center",
+                    verticalalignment="top",
+                    fontsize="small",
+                )
+                data["fig_scan_Phi"] = plt.gcf()
+                data["ax_scan_Phi"] = plt.gca()
+                plt.figure(figsize=(16, 12))
+                for whichPlot in range(numPlots):
+                    plt.subplot(numRows, numCols, whichPlot + 1)
+                    Bn = Bn_arrs[ialpha_to_plot[whichPlot] - 1]
+
+                    plt.rcParams.update({"font.size": 18})
+
+                    plt.contourf(
+                        eval_grid.nodes[eval_grid.unique_zeta_idx, 2],
+                        eval_grid.nodes[eval_grid.unique_theta_idx, 1],
+                        (Bn).reshape(
+                            eval_grid.num_theta, eval_grid.num_zeta, order="F"
+                        ),
+                        levels=ncontours,
+                    )
+                    plt.ylabel("theta")
+                    plt.xlabel("zeta")
+                    plt.title(
+                        f"alpha= {alphas[ialpha_to_plot[whichPlot] - 1]:1.5e}"
+                        + f" index = {ialpha_to_plot[whichPlot] - 1}",
+                        fontsize="x-small",
+                    )
+                    plt.colorbar()
+                    plt.xlim([0, 2 * np.pi / eq.NFP])
+                plt.tight_layout()
+                units = " (T)" if not normalize else " (unitless)"
+                plt.figtext(
+                    0.5,
+                    0.995,
+                    "Bnormal" + units,
+                    horizontalalignment="center",
+                    verticalalignment="top",
+                    fontsize="small",
+                )
+                data["fig_scan_Bn"] = plt.gcf()
+                data["ax_scan_Bn"] = plt.gca()
+
+        data["alpha"] = alphas[0] if not scan else alphas
+        data["Phi_mn"] = phi_mns[0] if not scan else phi_mns
+        data["I"] = I
+        data["G"] = G
+        data["chi^2_B"] = chi2Bs[0] if not scan else chi2Bs
+        data["chi^2_K"] = chi2Ks[0] if not scan else chi2Ks
+        data["|K|"] = K_mags[0] if not scan else K_mags
+
+        self.Phi_mn = phi_mns[0]
+
+        return data
+
+    def cut_surface_current_into_coils(  # noqa: C901 - FIXME: simplify this
+        self,
+        desirednumcoils=10,  # TODO: make this coils_per_NFP for modular...
+        step=2,
```

Comment:
it can sort of smooth the curve out, and make the resulting SplineXYZCoil object less expensive to compute. Might not be needed though, I can test if it makes much difference and remove if not but I remember it being useful before for me